androidsdk/src/main/java/com/wootric/androidsdk/utils/SHAUtil.java

```diff
@@ -0,0 +1,46 @@
+package com.wootric.androidsdk.utils;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+
+/**
+ * Created by diegoserranoa on 4/25/16.
+ */
+public final class SHAUtil {
+
+    final static String POSSIBLE = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+    private SHAUtil(){}
+
+    public static String buildUniqueLink(String accountToken, String endUserEmail) {
+        Date date = new Date();
+        String randomText = randomString();
+
+        String text = accountToken + endUserEmail + date.toString() + randomText;
+
+        MessageDigest mDigest = null;
+
+        StringBuffer sb = new StringBuffer();
+        try {
+            mDigest = MessageDigest.getInstance("SHA256");
+            byte[] result = mDigest.digest(text.getBytes());
+            for (int i = 0; i < result.length; i++) {
+                sb.append(Integer.toString((result[i] & 0xff) + 0x100, 16).substring(1));
```

Comment:
I found this library from Apache which would make it simpler 
[http://mvnrepository.com/artifact/commons-codec/commons-codec/1.8](url)